### ISSUE TITLE
Support for different Samsung Browser versions

### ISSUE THREAD

**Issue by max-b** [rofrischmann/inline-style-prefixer#146]
Testing a browser with the following user agent string:
`Mozilla/5.0 (Linux; Android 5.0.2; SAMSUNG-SM-G850A Build/LRX22G) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/2.1 Chrome/34.0.1847.76 Mobile Safari/537.36`

I get back this from https://github.com/rofrischmann/inline-style-prefixer/blob/master/modules/utils/getBrowserInformation.js
```
{"name":"Samsung Internet for Android","samsungBrowser":true,"version":"2.1","blink":true,"android":true,"osname":"Android","osversion":"5.0.2","mobile":true,"x":true,"jsPrefix":"Webkit","cssPrefix":"-webkit-","browserName":"and_chr","browserVersion":44,"osVersion":5}
```
Related to https://github.com/rofrischmann/inline-style-prefixer/issues/102
On https://github.com/rofrischmann/inline-style-prefixer/blob/master/modules/utils/getBrowserInformation.js#L110 the library is setting every samsung browser to `and_chr` version 44. What it looks like to me is that the above userAgent string should actually be set to `and_chr` version 34. As an additional piece of evidence to support that, the browser with that userAgent string is not displaying `transform` styles correctly because it requires a `-webkit` prefix. As per http://caniuse.com/#search=transform Chrome 44 does not require a prefix, but Chrome 34 *does*.

`bowser` returns this from the above userAgent string:
```
{"name":"Samsung Internet for Android","samsungBrowser":true,"version":"2.1","blink":true,"android":true,"osname":"Android","osversion":"5.0.2","mobile":true,"x":true,"jsPrefix":"Webkit","cssPrefix":"-webkit-","browserName":"android","browserVersion":2.1,"osVersion":5}
```

Similarly, if I use this userAgent string:
`Mozilla/5.0 (Linux; Android 6.0.1; SAMSUNG SM-G900F Build/MMB29M) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/4.0 Chrome/44.0.2403.133 Mobile Safari/537.36`

`bowser` returns:
```
{"name":"Samsung Internet for Android","samsungBrowser":true,"version":"4.0","blink":true,"android":true,"osname":"Android","osversion":"6.0.1","mobile":true,"a":true}
```
Which is unfortunate, because what it looks to me is that actually you would want to get the `Chrome` build version, which it doesn't seem like `bowser` supplies. 

It looks like there is a predictable relationship between Samsung Browser versions and Chrome versions:
https://developers.whatismybrowser.com/useragents/explore/software_name/samsung-browser/?order_by=operating_system_name

I'm not sure if you think the mapping is appropriate in this library, but it looks like it should be:
Samsung Browser -> Chrome
7 -> 59
6 -> 56
5 -> 51
4 -> 44
3 -> 38
2 -> 34
1 -> ?? (unclear - it seems like it's rare/mostly for smart tvs??)

Would you accept a PR which just did that mapping inside `getBrowserInformation`?

**Comment by rofrischmann** [comment]
Hmm I am not totally sure about this. Isn't this something we would want to get from bowser in the first hand?

For this library, I would love to only process the userAgent information from bowser (in a perfect world) but we're already doing some validation which I don't like at all.

**Comment by max-b** [comment]
Yeah I agree that this is not a great solution. It actually looks like `caniuse` supports 'samsung' as a browser type, but only back to version #4:
https://github.com/Fyrd/caniuse/blob/8c9d2d0e7f1fa9df091b5886e60862acfcf7d599/features-json/transforms2d.json#L317

It seems to me that in a perfect world, that would be the place to address this...
